Current master build fails on LuaJit

*Title*: The current build of master fails.

*Description*:
I pulled the latest master and tried to build.
After a `bazel clean --expunge` the build breaks at a very early stage.

Building with `bazel build //source/exe:envoy-static` gives this error:
```
==== Building LuaJIT 2.1.0-beta3 ====
make -C src
make[1]: Entering directory '/home/robert/.cache/bazel/_bazel_robert/f53035c79fe3a38f252a2a5d1464c7c1/external/com_github_luajit_luajit/src'
HOSTCC    host/minilua.o
/home/robert/.cache/bazel/_bazel_robert/f53035c79fe3a38f252a2a5d1464c7c1/external/local_config_cc/extra_tools/envoy_cc_wrapper  -O2 -fomit-frame-pointer -Wall  -DLUAJIT_ENABLE_LUA52COMPAT  -I. -DLUAJIT_TARGET=LUAJIT_ARCH_x64 -DLJ_ARCH_HASFPU=1 -DLJ_ABI_SOFTFP=0  -c -o host/minilua.o host/minilua.c
Assembler messages:
Fatal error: can't create host/minilua.o: Read-only file system
Makefile:691: recipe for target 'host/minilua.o' failed
make[1]: *** [host/minilua.o] Error 1
make[1]: Leaving directory '/home/robert/.cache/bazel/_bazel_robert/f53035c79fe3a38f252a2a5d1464c7c1/external/com_github_luajit_luajit/src'
Makefile:112: recipe for target 'src/luajit' failed
make: *** [src/luajit] Error 2
cp: cannot stat '/tmp/tmp.C83bga9crU/./luajit': No such file or directory
```

The full build log is here:
[build.log](https://github.com/envoyproxy/envoy/files/2935735/build.log)

--- a/bazel/foreign_cc/luajit.patch
+++ b/bazel/foreign_cc/luajit.patch
@@ -48,17 +48,20 @@ new file mode 100755
 index 0000000..9c71271
 --- /dev/null
 +++ b/build.py
-@@ -0,0 +1,25 @@
+@@ -0,0 +1,28 @@
 +#!/usr/bin/env python
 +
 +import argparse
 +import os
++import shutil
 +
 +def main():
 +    parser = argparse.ArgumentParser()
 +    parser.add_argument("--prefix")
 +    args = parser.parse_args()
-+    os.chdir(os.path.dirname(os.path.realpath(__file__)))
++    src_dir = os.path.dirname(os.path.realpath(__file__))
++    shutil.copytree(src_dir, os.path.basename(src_dir))
++    os.chdir(os.path.basename(src_dir))
 +
 +    os.environ["MACOSX_DEPLOYMENT_TARGET"] = "10.6"
 +    os.environ["DEFAULT_CC"] = os.environ.get("CC", "")

